syntax = "proto3";

package flip;

import "flip_spec.proto";

message FlipListResponse {
  message FlipInfo { string info = 1; }

  repeated FlipInfo infos = 1;
}

message FlipNameRequest { string name = 1; }

message FlipRemoveRequest { string name = 1; }

message FlipRemoveResponse { uint32 num_removed = 1; }

service FlipServer {
  // Inject a fault rpc
  rpc InjectFault(flip.FlipSpec) returns (flip.FlipResponse);

  // Get details about one or all faults
  rpc GetFaults(FlipNameRequest) returns (FlipListResponse);

  // Remove a fault added earlier
  rpc RemoveFault(FlipRemoveRequest) returns (FlipRemoveResponse);
}